### DISCUSSION TITLE
Character encoding

### DISCUSSION THREAD

**Discussion by binerf** [photoprism/photoprism#3567] [Q&A] [answered]
Hello,

With the last version of Photoprism (docker version), I'm facing issue with character encoding.
Title of imported photos containing character with "accent" are not well encoded.

Ex: Vacances Ã©tÃ© 2022 (should be _Vacances été 2022_)

Do I have to configure something somewhere or any setting to change ?

**Comment by lastzero** [comment]
From what version did you upgrade? Is it displayed correctly when you use Exiftool? Is it encoded in UTF8 (unicode)?

**Reply by lastzero** [reply]
We've been using Unicode from the start and haven't changed anything related to this, so there's also no config option you could change.

**Reply by binerf** [reply]
Hello,
Actually, I started from scratch with Build [230719-73fa7bbe8] (found in setting page)

With Exiftool on my laptop:
Title                           : Vacances été 2022

How to check which encoding is currently in use ?

**Reply by lastzero** [reply]
There's different ways to do that, depending on your operating systems and tools available. I would recommend to follow the advice and try the options in the Exiftool FAQ: https://exiftool.org/faq.html#Q10

**Reply by binerf** [reply]
But, as I'm using docker, it should be configured in the docker image, right ?

I've checked with exiftool directly connected in Photoprim container and same result.
Title                           : Vacances été 2022

**Reply by lastzero** [reply]
Unicode is the same for all languages. There is no need for a configuration. We found that sometimes text is encoded with ISO or Windows code pages, especially when they were edited on Windows. Since our team has very limited time, it would be great if you could verify that before filing a bug report.

**Reply by binerf** [reply]
Maybe I can provide with a photo which cause this issue ?

**Reply by graciousgrey** [reply]
Sure, you can send us a sample file to samples@photoprism.app :)

**Reply by binerf** [reply]
OK sent.

**Reply by graciousgrey** [reply]
Thank you for providing us with the file. Exiftool shows the following information:

``` 
  "Caption": "Vacances été 2022",
  "Title": "Vacances été 2022",
  "Caption-Abstract": "Vacances Ã©tÃ© 2022",
  "Headline": "Vacances Ã©tÃ© 2022",
````

PhotoPrism prioritizes the headline field over the title field (https://www.photoprism.app/kb/metadata). That's why the title in PhotoPrism is "Vacances Ã©tÃ© 2022". However this has not changed in recent versions. 
Do you know what other tool you are using is writing  those exif fields?

**Reply by binerf** [reply]
Oh sorry didn't check this information about headline field.
Thank you for informing.

To create those exif information, I use [Shotwell ](https://wiki.gnome.org/Apps/Shotwell)(laptop with Linux Mint).

**Reply by graciousgrey** [reply]
Maybe there are some settings in Shotwell that prevent writing strings like  "Vacances Ã©tÃ© 2022"

**Reply by binerf** [reply]
Seems that IPTC fields don't handle easily UTF8 (https://exiftool.org/TagNames/IPTC.html)

**Reply by lastzero** [reply]
Interesting! In this case, it might be better to prioritize the other tags or add support for non-Unicode character sets (which won't be a quick win and therefore not available in the short term).

**Reply by lastzero** [reply]
![Screenshot_20230727_204655](https://github.com/photoprism/photoprism/assets/301686/a268632c-7946-45a7-a547-03a271e56355)

From my understanding, the JSON files created by Exiftool are invalid if (some) text is ISO encoded instead of unicode. Even mixing different unicode isn't possible unless I did it wrong my entire life.

**Comment by lastzero** [comment]
@binerf I've changed order of field names from which the title is extracted in commit https://github.com/photoprism/photoprism/commit/82dac4b7db65f1e490d3cd26a17b122832b0445f. It would be great if you could help us test this change using the development preview build we provide (or our public demo):

https://docs.photoprism.app/release-notes/#development-preview

Thank you!

**Reply by binerf** [reply]
Hi,

I've upgraded my instance with the "preview" docker image then did a full rescan of my library and I still have some characters not recognized.
![IMG_20230804_190027](https://github.com/photoprism/photoprism/assets/10645373/d03dcf1c-098f-41ed-a08f-68be25fca9ab)

The first line is now correct but the second one not yet.

**Reply by lastzero** [reply]
It's the description, which is also extracted from one of the affected fields.

**Reply by binerf** [reply]
Is that possible to fix this field too?

Thanks 🙏

**Reply by lastzero** [reply]
Yes, of course. Simplest way would be to ignore the caption abstract field?

**Reply by graciousgrey** [reply]
Others might also use the caption-abstract field with characters that are not problematic and do not want to lose that information. We could alternatively prefer the caption over the caption-abstract for the description field.

**Reply by lastzero** [reply, marked as answer]
@binerf An updated development preview build will be available for you soon to test the changes!

**Reply by binerf** [reply]
It works!
Thank you!
![IMG_20230811_194536](https://github.com/photoprism/photoprism/assets/10645373/14ff3919-3268-4093-8ba6-6f276618fd90)

Do you think it could be possible to display the EXIF field named _User Comment_ instead of having twice the Title? 🤔

**Reply by lastzero** [reply]
I guess the description could be ignored if it's the same as the title...

**Reply by binerf** [reply]
Yes, but I find this line interesting to display. 
Maybe a configuration option to select EXIF information/field to display (could be Comment, Keywords, etc...) and which can be disabled.

**Reply by lastzero** [reply]
We can look into this if you can wait.

**Reply by binerf** [reply]
Many thanks!

Of course, I can wait; no rush at all. 

I already enjoy your work. Thanks for that!
